src/sagemaker/experiments/run.py

```diff
@@ -861,14 +861,15 @@ def list_runs(
             AWS services needed. If not specified, one is created using the
             default AWS configuration chain.
         max_results (int): Maximum number of Run objects to retrieve (default: None).
-        next_token (str): Token for next page of results (default: None).
         sort_by (SortByType): The property to sort results by. One of NAME, CREATION_TIME
             (default: CREATION_TIME).
         sort_order (SortOrderType): One of ASCENDING, or DESCENDING (default: DESCENDING).
 
     Returns:
         list: A list of ``Run`` objects.
     """
+
+    # all trial components retrieved by default
```

Comment:
You mean the `_TrialComponent.list` can retrieve all trial components?
https://github.com/aws/sagemaker-python-sdk/blob/8d2c16bb912f7a86f8598b1fa799fe6981e79dba/src/sagemaker/experiments/trial_component.py#L248-L262

Or the list trial component API can retrieve all TCs in one call?

If the next_token is not causing any issues, I'd suggest to keep it to avoid a breaking change

Reply:
all trial components are retrieved by default:

https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/apiutils/_base_types.py#L118-L147

The problem with this parameter is where is the customer going to get the next_token to supply to method in the first place? The only way to get next_token is to call `ListTrialComponents` in which case they have no reason to use this method. 

Also this method doesn't return a `NextToken`.

Reply:
I prefer to remove since it is not providing useful behavior as it currently is. In the future we could re-add it but will need to return a NextToken so the customer can supply it in their own page-loop behavior.